### PR TITLE
[OPERATOR-420] get px repo image from version websvc

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/apis/core/v1/storagecluster.go [hunk 1]
```diff
@@ -596,6 +596,7 @@ type ComponentImages struct {
 	Telemetry                  string `json:"telemetry,omitempty"`
 	MetricsCollector           string `json:"metricsCollector,omitempty"`
 	MetricsCollectorProxy      string `json:"metricsCollectorProxy,omitempty"`
+	PxRepo                     string `json:"pxrepo,omitempty"`
```
Comment (piyush-nimbalkar) [inline pkg/apis/core/v1/storagecluster.go:599]: nit: Should we call it `pxRepo` to be consistent?

Comment (piyush-nimbalkar) [inline pkg/apis/core/v1/storagecluster.go:599]: Can we add this to the schema under `deploy/crds/` so it doesn't get pruned by k8s as unknown field?

File: drivers/storage/portworx/util/util.go [hunk 2]
```diff
@@ -931,6 +931,12 @@ func IsTelemetryEnabled(spec corev1.StorageClusterSpec) bool {
 		spec.Monitoring.Telemetry.Enabled
 }
 
+// IsPxRepoEnabled returns true is telemetry is enabled
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/util/util.go:934]: nit: telemetry -> px repository

File: drivers/storage/portworx/portworx.go [hunk 3]
```diff
@@ -219,6 +219,13 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 			toUpdate.Status.DesiredImages.MetricsCollectorProxy = release.Components.MetricsCollectorProxy
 		}
 
+		if autoUpdatePxRepo(toUpdate) &&
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx.go:222]: Can we add unit tests for this? You can refer the tests here - https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/portworx_test.go#L575.

File: pkg/apis/core/v1/storagecluster.go [hunk 4]
```diff
@@ -90,7 +90,7 @@ type StorageClusterSpec struct {
 	// UserInterface contains details of a user interface for the storage driver
 	UserInterface *UserInterfaceSpec `json:"userInterface,omitempty"`
 	// PxRepo contains configuration for apt repository. Portworx uses it to install dependency modules.
-	PxRepo *PxRepoSpec `json:"pxrepo,omitempty"`
+	PxRepo *PxRepoSpec `json:"pxRepo,omitempty"`
```
Comment (piyush-nimbalkar) [inline pkg/apis/core/v1/storagecluster.go:93]: We will have to update the schema as well to `spec.pxRepo`

File: drivers/storage/portworx/portworx_test.go [hunk 5]
```diff
@@ -572,6 +572,121 @@ func TestStorageClusterDefaultsForLighthouse(t *testing.T) {
...
+	// some other component has changed
+	cluster.Spec.UserInterface = &corev1.UserInterfaceSpec{
+		Enabled: true,
+	}
+	driver.SetDefaultsOnStorageCluster(cluster)
+	require.Equal(t, "portworx/px-repo:old", cluster.Status.DesiredImages.PxRepo)
+
+	// Change desired image if px image is not set (new cluster)
+	cluster.Spec.Image = ""
+	cluster.Status.DesiredImages.Autopilot = "portworx/px-repo:old"
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx_test.go:642]: cluster.Status.DesiredImages.PxRepo

File: drivers/storage/portworx/portworx_test.go [hunk 6]
```diff
@@ -572,6 +572,121 @@ func TestStorageClusterDefaultsForLighthouse(t *testing.T) {
...
+	driver.SetDefaultsOnStorageCluster(cluster)
+	require.Empty(t, cluster.Spec.PxRepo.Image)
+	require.Equal(t, "portworx/px-repo:"+newCompVersion(), cluster.Status.DesiredImages.PxRepo)
+
+	// Change desired image if px image has changed
+	cluster.Spec.Image = "px/image:4.0.0"
+	cluster.Status.DesiredImages.Autopilot = "portworx/px-repo:old"
```
Comment (piyush-nimbalkar) [inline drivers/storage/portworx/portworx_test.go:649]: cluster.Status.DesiredImages.PxRepo

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM after minor fixes

**Comment by haibinxie** [comment]
@piyush-nimbalkar all comments resolved, merging it

**Pull request merged**